File: src/main/java/Pages/ProductPage.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;


public class ProductPage {
    By NameProd = By.cssSelector("h2[class='name']");
    By PriceProd = By.cssSelector("h3[class='price-container']");
    By AddToCart = By.cssSelector("a[onclick*='addToCart']");



    WebDriver driver;


    public ProductPage(WebDriver driver) {
        this.driver = driver;
    }

    public void AddTocart(){
        BasePage.WaitElement(AddToCart);
        driver.findElement(AddToCart).click();

    }

    public String PriceProd(){

        BasePage.WaitElement(PriceProd);
        String prodPrice = driver.findElement(PriceProd).getText();

        return prodPrice.substring(prodPrice.indexOf("$") + 1,prodPrice.indexOf(" "));
    }


    public String NameProd(){
        BasePage.WaitElement(NameProd);
        return driver.findElement(NameProd).getText();
    }






}

File: test/pageobjects/productPage.js
const Page = require('./page');


class productPage extends Page {


    get nameProd() {
        return $("h2[class='name']");
    }

    get priceProd() {
        return $("h3[class='price-container']");
    }

    get btnAddToCart() {
        return $("a[onclick*='addToCart']");
    }


    clickToAddToCart() {
        super.clicktoBtn(this.btnAddToCart);
    }


    getNameProd() {
        return this.nameProd.getText()
    }

    async getPriceProd() {
        const price = await this.priceProd.getText();
        const start = price.indexOf("$");
        const end = price.indexOf(" ");
        return price.substring(start+1, end);



    }

}

module.exports = new productPage();